docs/concepts/composition.md

```diff
@@ -0,0 +1,134 @@
+---
+title: Composing environments
+description: How to combine and reuse environments.
+---
+
+# Composing environments
+
+When your company starts work on a new service, it's likely that the tools required to work on this service will be very similar to the tools used to work on other services at the company.
+Similarly, when you work on hobby projects it's likely that you use a similar set of tools from one project to the next.
+With other developer environment solutions it's common to need to need to recreate your developer environment from scratch for each new project.
+
+With Flox you can define a toolchain or service once and _compose_ it with others to create a composed environment.
+Put another way, **with Flox you can create developer environments from reusable building blocks**.
+Reuse and composition are two features that have long been a Holy Grail for developer environments, and Flox makes these features friendly and easily accessible.
+
+## Building blocks
+
+Environments are the building blocks from which a "composed" environment is created.
+In this composition there is a hierarchy consisting of a "composing" environment and a list of "included" environments that it treats as dependencies.
+The manifests of the included and composing environments are merged and then the resulting "merged" manifest is locked and built.
+
+```d2 scale="0.9"
+composer: "Composing\nenvironment"
+
+includes: Included environments {
+  envA: Env A
+  envB: Env B
+  envC: Env C
+  direction: down
+  envA -> envB: Overridden by
+  envB -> envC: Overridden by
+}
+
+composer -> includes: Overrides
```

Comment:
This flow feels more natural to me because it's all in the same direction: 
![image](https://github.com/user-attachments/assets/3ec81642-a395-45a2-bad3-e97791e8c67b)

```suggestion
includes -> composer: Overridden by
```

Reply:
Makes sense